### PR TITLE
Cat devel

### PR DESCRIPTION
**Pull request by vcatlett-nrao**
Big dump of my stuff, including GUI code, documentation, repaired GitHub Actions, and a bug fix for that SDFITS unit warning (added `unit_parse_strict='silent'` to` Table.read()` in `SDFITSLoad.creat_index()` )

### CODE REVIEW

File: src/dysh/spectra/tests/test_scan.py
Comment (mpound) [inline src/dysh/spectra/tests/test_scan.py:1]: Why was this changed to assert True?

File: threading/dev
Comment (mpound) [inline threading/dev:1]: What is in the threading directory?   What is in the binary files and why is there a C++ file?

File: src/dysh/spectra/tests/test_scan.py
Comment (vcatlett-nrao) [reply src/dysh/spectra/tests/test_scan.py:1]: Note in the comments above that line, it says "this won't pass yet" because there's no established threshold for what's an ok difference between the spectra. The original assert was that the mean difference was exactly 0, which would never pass. I left the original assert in a comment and then forced the test to pass so that the pytest results wouldn't always be a failure.

File: threading/dev
Comment (vcatlett-nrao) [reply threading/dev:1]: Something that was supposed to stay in my local folders, oops. I'm working on various threading techniques to speed up the GUI. Let me know if I need to cancel this pull request, remove the folder, than request a new one or if it's possible to just reject that folder.

File: src/dysh/fits/sdfitsload.py
Comment (mpound) [inline src/dysh/fits/sdfitsload.py:1]: I am in the middle of large changes to sdfitsload/gbtfitsload and I am concerned about dealing with the fallout of your large changes when I have to merge.

File: docs/source/conf.py
Comment (mpound) [inline docs/source/conf.py:1]: why was numpydoc removed?

File: src/dysh/fits/sdfitsload.py
Comment (vcatlett-nrao) [reply src/dysh/fits/sdfitsload.py:1]: Fair, it's on me for waiting so long to add my changes. Let me close this request and make a dedicated branch for this merge migration and do pieces one at a time. I'll keep all my sdfitsload changes to myself except for that unit warning fix.

**Comment by vcatlett-nrao** [comment]
This was clearly too big of a dump to try at once. I'm closing this request and trying smaller ones.
